cli/src/main.rs

```diff
@@ -38,7 +38,15 @@ fn main() {
 
     setup_panic_hooks();
 
-    let mut session = Session::init().unwrap();
+    let session_result = Session::init();
+
+    if let Err(err) = session_result {
+        report(&err);
+        let code = err.exit_code();
+        return code.exit()
```

Comment:
Welp, this sucks 🤷 

```
warning: unreachable expression
  --> cli/src/main.rs:46:9
   |
46 |         return code.exit()
   |         ^^^^^^^-----------
   |         |      |
   |         |      any code following this expression is unreachable
   |         unreachable expression
   |
   = note: `#[warn(unreachable_code)]` on by default
```

Sorry about the back and forth but maybe we have to live with `code.exit()` and no visual indication. Maybe add an obvious comment?
```rust
// XXX: THIS LINE QUITS THE PROGRAM
```

Reply:
@Enrico2 another solution we might like to adopt is would be this:

https://stackoverflow.com/a/30285110

```rs
fn main() {
    let exit_code = real_main();
    std::process::exit(exit_code);
}

fn real_main() -> i32 {
    // the real program here
    if let Err(err) = session_result {
        report(&err);
        return err.exit_code();
    }
    //...
}
```

I think this potentially sets us up for better unit testing as well, and gets us all the visual diffing we're after. I'd be happy to go this route as it's a rather quick change.